aten/src/ATen/native/cpu/MaxPoolKernel.cpp

```diff
@@ -25,9 +27,8 @@ void cpu_max_pool(
   auto input = input_.contiguous();
   auto output = output_.contiguous();
   auto indices = indices_.contiguous();
-
-  auto input_data = input.data_ptr<scalar_t>();
-  auto output_data = output.data_ptr<scalar_t>();
+  auto input_data = std::is_integral<scalar_t>::value ? reinterpret_cast<scalar_t*>(input.data_ptr()) : input.data_ptr<scalar_t>();
```

Comment:
You're trying very hard not to add a template parameter but it is resulting in code that doesn't make sense. If I didn't know anything about quantization, WHY would I expect an integral `scalar_t` to require reinterpret casting from data ptr. It doesn't make any sense! At the very least there should be a comment but it would be better to structure the code in such a way that people would think "oh yeah, that makes sense." In particular, what is the valid `data_ptr` to extract for quantized types, and why can't it be used here?

Reply:
scalar_t here is probably things like uint8, and input.data_ptr() would have type quint8*, they have the same underlying type but we don't use quint8 for computations. can we do input.data_ptr<uint8*>() now btw, if so we can just remove this cast?

Reply:
thanks for the suggestions! yeah we want to keep `quint8` and `uint8` separate since unlike `uint8`, `quint8` in itself doesn't correspond to any values (doesn't have a meaning). We have done the first point you mentioned